### PR TITLE
Frontend shutdown before snapshot revert

### PR DESCRIPTION
**Pull request by shreesha21**
The following changes include two things:
1. Integration test to manually shutdown before snapshot revert operation and start the launcher after snapshot revert completion.
2. Modify the controller logic in revert to check if the frontend is disabled before performing snapshot revert. Otherwise, abort the snapshot revert.

### CODE REVIEW

File: controller/revert.go [hunk 1]
```diff
@@ -62,20 +60,6 @@ func (c *Controller) Revert(name string) error {
...
-
-		if c.launcher != "" {
-			logrus.Infof("Asking the launcher to start the frontend")
-			if err := c.launcherStartFrontend(); err != nil {
-				if !strings.Contains(err.Error(), "no frontend started since it's not specified") {
-					return err
-				}
-			}
-		}
-	}
```
Comment (shreesha21) [inline controller/revert.go]: This change is required, without which every single test case related to snapshot revert fails in integration test.
Hence, add the changes similar to the frontend restriction in backup restore.

File: integration/data/common.py [hunk 2]
```diff
@@ -151,6 +152,12 @@ def restore_with_no_frontend(backup, grpc_c):
     assert v.frontendState == "up"
 
 
+def snapshot_revert_with_no_frontend(name):
```
Comment (yasker) [inline integration/data/common.py]: This is in fact `snapshot_revert_with_frontend`.

File: integration/data/test_restore_to_file.py [hunk 3]
```diff
@@ -140,7 +142,15 @@ def restore_to_file_with_backing_file_test(backing_dev, backup_target):
     os.remove(output_qcow2_path)
     assert not os.path.exists(output_qcow2_path)
 
+    launcher.shutdown_engine_frontend()
```
Comment (yasker) [inline integration/data/test_restore_to_file.py]: Can we put those wrappers around `cmd.snapshot_revert` into a common function as well?

File: integration/data/common.py (hunk 2 above)
Comment (shreesha21) [reply integration/data/common.py]: The function disables the engine frontend before running snapshot_revert right?
Are you implying me to change it to `snpashot_revert_with_frontend_disable` ?

Comment (yasker) [reply integration/data/common.py]: I meant, this function disables the frontend, but it runs with frontend enabled. From the caller point of view, it's running the function when the frontend is enabled. The function knows what to do when the frontend is enabled (which is to disable it first). How to prepare the revert is not the concern of the caller.

Comment (shreesha21) [reply integration/data/common.py]: Agreed, will change this to `snapshot_revert_with_frontend`

**Pull request merged**